controllers/cleanup.go

```diff
@@ -178,7 +176,10 @@ func (r *VolumeSnapshotRestoreReconciler) CleanRestoreResources(log logr.Logger)
 	}
 
 	// make sure VSR is completed before deleting resources
-	if vsr.Status.Phase != volsnapmoverv1alpha1.SnapMoverRestoreVolSyncPhaseCompleted {
+	if vsr.Status.Phase != volsnapmoverv1alpha1.SnapMoverRestoreVolSyncPhaseCompleted &&
+		vsr.Status.Phase != volsnapmoverv1alpha1.SnapMoverRestorePhaseFailed &&
+		vsr.Status.Phase != volsnapmoverv1alpha1.SnapMoverRestorePhasePartiallyFailed &&
+		vsr.DeletionTimestamp.IsZero() {
 		r.Log.Info(fmt.Sprintf("waiting for volsync to complete before deleting volumesnapshotrestore %s resources", r.req.NamespacedName))
 		return false, nil
 	}
```

Comment:
```suggestion
	if vsr.Status.Phase != volsnapmoverv1alpha1.SnapMoverRestoreVolSyncPhaseCompleted &&
		vsr.Status.Phase != volsnapmoverv1alpha1.SnapMoverRestorePhaseFailed &&
		vsr.Status.Phase != volsnapmoverv1alpha1.SnapMoverRestorePhasePartiallyFailed &&
		!vsr.DeletionTimestamp.IsZero() {
		r.Log.Info(fmt.Sprintf("waiting for volsync to complete before deleting volumesnapshotrestore %s resources", r.req.NamespacedName))
		return false, nil
	}
```

It should be when deletionTimestamp is present when we would wanna log about `deleting volumesnapshotrestore %s resources` right?

Reply:
if you agree with my comments then please also fix the VSBs as well.

Reply:
No this log is for when the CR is not being deleted i.e. everything is working normally and we are just waiting for volsync to complete. IMO it is maybe an unnecessary log, but could be useful for debugging.